x/qgb/orchestrator/orchestrator_client.go

```diff
@@ -103,8 +110,65 @@ func (oc *orchestratorClient) SubscribeValset(ctx context.Context) (<-chan types
 	return valsetsChan, nil
 }
 
+func (oc *orchestratorClient) addOldVSAttestations(ctx context.Context, valsetsChan chan types.Valset) error {
+	oc.logger.Info("Started adding Valsets attestation to queue")
+	lastUnbondingHeight, err := oc.querier.QueryLastUnbondingHeight(ctx)
+	if err != nil {
+		oc.logger.Error(err.Error())
+		return err
+	}
+
+	valsets, err := oc.querier.QueryLastValsets(ctx)
+	if err != nil {
+		oc.logger.Error(err.Error())
+		return err
+	}
+
+	// todo: double check that the first validator set is found
+	if len(valsets) < 1 {
+		oc.logger.Error("no validator sets found")
+		return nil
+	}
+	valsetsChan <- valsets[0]
+
+	previousNonce := valsets[0].Nonce
+	for {
+		if previousNonce == 1 {
+			oc.logger.Info("Finished adding Valsets attestation to queue")
+			return nil
+		}
+		previousNonce = previousNonce - 1
+		lastVsConfirm, err := oc.querier.QueryValsetConfirm(ctx, previousNonce, oc.orchestratorAddress)
+		if err != nil {
+			oc.logger.Error(err.Error())
+			return err
+		}
+		// The valset signed by the orchestrator to get lastVsConfirm
+		// Used to get the height that valset waas first introduced
+		correspondingVs, err := oc.querier.QueryValsetByNonce(ctx, previousNonce)
+		if err != nil {
+			oc.logger.Error(err.Error())
+			return err
+		}
+		if correspondingVs.Height < lastUnbondingHeight {
+			// Most likely, we're up to date and don't need to catchup anymore
+			oc.logger.Info("Finished adding Valsets attestation to queue")
+			return nil
+		}
+		if lastVsConfirm != nil {
+			// in case we have holes in the signatures
+			continue
+		}
+
+		valsetsChan <- *correspondingVs
+	}
+}
+
 func (oc *orchestratorClient) SubscribeDataCommitment(ctx context.Context) (<-chan ExtendedDataCommitment, error) {
-	dataCommitments := make(chan ExtendedDataCommitment)
+	dataCommitments := make(chan ExtendedDataCommitment, 100)
```

Comment:
any particular reason for the 100 here other than that we want a buffered channel?

Reply:
I added it so that when replaying old attestations, and we're so far behind,  not to have a huuuuge channel.
Do you think we should remove it ?

Reply:
no I think that makes sense, it shouldn't be that much of a burden, and we can optimize later if we really want to "go golfing"